stats/json/collector.go

```diff
@@ -34,20 +37,18 @@ import (
 )
 
 type Collector struct {
-	outfile     io.WriteCloser
+	closeFn     func() error
 	fname       string
 	seenMetrics []string
-}
 
-// Verify that Collector implements lib.Collector
-var _ lib.Collector = &Collector{}
+	encoder *json.Encoder
 
-// Similar to ioutil.NopCloser, but for writers
-type nopCloser struct {
-	io.Writer
+	buffer     []stats.Sample
+	bufferLock sync.Mutex
 }
 
-func (nopCloser) Close() error { return nil }
+// Verify that Collector implements lib.Collector
+var _ lib.Collector = &Collector{}
 
 func (c *Collector) HasSeenMetric(str string) bool {
 	for _, n := range c.seenMetrics {
```

Comment:
Just looking at this, it seems like using a set (i.e. map) instead of a slice would be better?

Reply:
didn't seem to change much ... probably a sorted list using sort.Search will be faster, but  ... the json encoding is so much more expensive(the whole HandleMetric takes around 100ms for each 10 seconds that the encode takes) , that I decided against trying to optimize it.
Maybe given the fact I am doing changes I can move it to a map as at least that is somewhat easier to read.